src/identity.zig

```diff
@@ -77,27 +54,73 @@ pub fn getOrInit(allocator: std.mem.Allocator, logger: Logger) !KeyPair {
             defer file.close();
 
             const keypair = KeyPair.generate();
-            try file.writeAll(&keypair.secret_key.toBytes());
+            try std.json.stringify(&keypair.secret_key.toBytes(), .{}, file.writer());
 
             return keypair;
         },
     }
 }
 
-/// Reads a binary keypair file and returns its public key as `Pubkey`.
-pub fn readBinaryKeypairPubkey(path: []const u8) !sig.core.Pubkey {
-    var file = try std.fs.cwd().openFile(path, .{});
+/// Reads a pubkey either as a base58 string or from a json keypair file
+pub fn readPubkeyFlexible(logger: Logger, base58_or_kp_path: []const u8) !Pubkey {
+    const pk_err = if (Pubkey.parseRuntime(base58_or_kp_path)) |p| return p else |e| e;
+
+    errdefer |e| logger.err().logf(
+        "Could not interpret '{s}' as a base58 pubkey due to {} or a json keypair file due to {}",
+        .{ base58_or_kp_path, pk_err, e },
+    );
```

Comment:
`errdefer |e|` is due to be removed as a feature, so I would avoid using it here.